scripts/020_cluster_and_select_design_points.R

```diff
@@ -0,0 +1,264 @@
+library(ggplot2)
+# settings
+source("000-config.R")
+
+#' ## Load Site Environmental Data Covariates
+#'
+#' Environmental data was pre-processed in the previous workflow 00-prepare.qmd.
+#'
+#' Below is a summary of the covariates dataset
+#'
+#' - site_id: Unique identifier for each polygon
+#' - crop: Crop type
+#' - pft: Plant functional type
+#' - temp: Mean Annual Temperature from ERA5
+#' - precip: Mean Annual Precipitation from ERA5
+#' - srad: Solar Radiation
+#' - vapr: Vapor pressure deficit
+#' - clay: Clay content from SoilGrids
+#' - ocd: Organic Carbon content from SoilGrids
+#' - twi: Topographic Wetness Index
+#' - climregion_id: identifier for each climate region
+selected_covariates <- c("temp", "precip", "srad", "vapr", "clay", "ocd", "twi")
+
+ca_fields <- sf::st_read(file.path(data_dir, "ca_fields.gpkg"))
+
+site_covariates_csv <- file.path(data_dir, "site_covariates.csv")
+site_covariates <- readr::read_csv(site_covariates_csv) |>
+  dplyr::left_join(ca_fields  |> dplyr::select(site_id, lat, lon), by = "site_id")
```

Comment:
I've been finding that pipes are visually louder than parens, making it easy to mis-skim this as "left join, then select, then -- wait, what's this 'by' doing over here". Indenting may make it clearer:
```suggestion
  dplyr::left_join(
    ca_fields  |> dplyr::select(site_id, lat, lon),
    by = "site_id")
```

Some days I even lean toward skipping the pipe and writing `...join(dplyr::select(ca_fields, site_id, lat, lon), by =...)`, but I might be in a minority there.